inc/class-sitemaps-pages.php

```diff
@@ -11,28 +11,33 @@ class Core_Sitemaps_Pages extends Core_Sitemaps_Provider {
 	 * @var string
 	 */
 	protected $object_type = 'page';
+
 	/**
 	 * Sitemap name
+	 *
 	 * Used for building sitemap URLs.
 	 *
 	 * @var string
 	 */
-	protected $name = 'pages';
+	public $name = 'pages';
 
 	/**
-	 * Bootstrapping the filters.
+	 * Sitemap route.
+	 *
+	 * Regex pattern used when building the route for a sitemap.
+	 *
+	 * @var string
 	 */
-	public function bootstrap() {
-		add_action( 'core_sitemaps_setup_sitemaps', array( $this, 'register_sitemap' ), 99 );
-		add_action( 'template_redirect', array( $this, 'render_sitemap' ) );
-	}
+	public $route = '^sitemap-pages\.xml$';
 
 	/**
-	 * Sets up rewrite rule for sitemap_index.
+	 * Sitemap slug.
+	 *
+	 * Used for building sitemap URLs.
+	 *
+	 * @var string
 	 */
-	public function register_sitemap() {
-		$this->registry->add_sitemap( $this->name, '^sitemap-pages\.xml$', esc_url( $this->get_sitemap_url( $this->name ) ) );
-	}
+	public $slug = 'page';
```

Comment:
```suggestion
	public $slug = 'pages';
```
Does not match $route, and the post provider is using plural here.